src/controller/python/chip/commissioning/__init__.py

```diff
@@ -0,0 +1,142 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from typing import *
+
+import abc
+import dataclasses
+import enum
+import os
+
+from chip import clusters as Clusters
+
+
+ROOT_ENDPOINT_ID = 0
+
+
+@dataclasses.dataclass
+class CommissioneeInfo:
+    endpoints: Set[int]
+    is_thread_device: bool = False
+    is_wifi_device: bool = False
+    is_ethernet_device: bool = False
+
+
+class RegulatoryLocationType(enum.IntEnum):
+    INDOOR = 0
+    OUTDOOR = 1
+    INDOOR_OUTDOOR = 2
+
+
+@dataclasses.dataclass
+class RegulatoryConfig:
+    location_type: RegulatoryLocationType
+    country_code: str
+
+
+@dataclasses.dataclass
+class PaseParameters:
+    setup_pin: int
+    temporary_nodeid: int
+
+
+@dataclasses.dataclass
+class PaseOverBLEParameters(PaseParameters):
+    discriminator: int
+
+    def __str__(self):
+        return f"BLE:0x{self.discriminator:03x}"
+
+
+@dataclasses.dataclass
+class PaseOverIPParameters(PaseParameters):
+    address: str
```

Comment:
As mentioned previously, we should use DNS-SD results (or discriminator + passcode) rather than direct IP address. All "direct IP address" APIs are deprecated.